habitat-lab/habitat/tasks/nav/nav.py

```diff
@@ -1289,16 +1402,553 @@ def step(
+
+    def step(self, *args: Any, **kwargs: Any):
+        r"""Update ``_metric``, this method is called from ``Env`` on each
+        ``step``.
+        """
+        xyt_waypoint = np.array([0.0, 0.0, self._turn_angle])
+        return self._step_rel_waypoint(
+            xyt_waypoint, self._config.max_wait_duration, *args, **kwargs
+        )
+
+    @property
+    def action_space(self):
+        return EmptySpace()
+
+
+@registry.register_task_action
+class TurnRightWaypointAction(WaypointAction):
+    name: str = "turn_right_waypoint"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self._turn_angle = self._config.turn_angle
+        self._max_wait_duration = self._config.max_wait_duration
+
+    def step(self, *args: Any, **kwargs: Any):
+        r"""Update ``_metric``, this method is called from ``Env`` on each
+        ``step``.
+        """
+        xyt_waypoint = np.array([0.0, 0.0, -self._config.turn_angle])
+        return self._step_rel_waypoint(
+            xyt_waypoint, self._config.max_wait_duration, *args, **kwargs
+        )
+
+    @property
+    def action_space(self):
+        return EmptySpace()
+
+
+@registry.register_task_action
+class LookUpDiscreteToVelocityAction(WaypointAction):
+    name: str = "look_up_discrete_to_velocity"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self._turn_angle = self._config.turn_angle
+        self._max_wait_duration = self._config.max_wait_duration
+
+    def step(self, *args: Any, **kwargs: Any):
+        r"""Update ``_metric``, this method is called from ``Env`` on each
+        ``step``.
+        """
+        xyt_waypoint = np.array([0.0, 0.0, 0.0])
+        delta_camera_pitch_angle = self._turn_angle
+        return self._step_rel_waypoint(
+            xyt_waypoint,
+            self._config.max_wait_duration,
+            delta_camera_pitch_angle,
+            *args,
+            **kwargs,
+        )
+
+    @property
+    def action_space(self):
+        return EmptySpace()
+
+
+@registry.register_task_action
+class LookDownDiscreteToVelocityAction(WaypointAction):
+    name: str = "look_down_discrete_to_velocity"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self._turn_angle = self._config.turn_angle
+        self._max_wait_duration = self._config.max_wait_duration
+
+    def step(self, *args: Any, **kwargs: Any):
+        r"""Update ``_metric``, this method is called from ``Env`` on each
+        ``step``.
+        """
+        xyt_waypoint = np.array([0.0, 0.0, 0.0])
+        delta_camera_pitch_angle = -self._turn_angle
+        return self._step_rel_waypoint(
+            xyt_waypoint,
+            self._config.max_wait_duration,
+            delta_camera_pitch_angle,
+            *args,
+            **kwargs,
+        )
+
+    @property
+    def action_space(self):
+        return EmptySpace()
```

Comment:
Why it's EmptySpace?

Reply:
Because the action type has no parameters. Like calling `MoveFowardWaypointAction` always moves the robot forward by the amount specified in configs.

Personally I think it makes more sense to define a `CameraDiscreteAction` and have parameters specify whether you want to look down, look up, etc. But I don't know habitat well enough so I just went with whatever's there.

Reply:
@exhaustin, if that's the case, would you consider leaving a TODO note?